### PR TITLE
feat(route): migrate from request interception w/ events to page.route

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -2901,14 +2909,13 @@ page.on('requestfailed', request => {
 - returns: <[Promise]>
 
 Fulfills request with given response. To use this, request interception should
-be enabled with `page.setRequestInterception`. Exception is thrown if
+be enabled with `page.route`. Exception is thrown if
```
Comment (yury-s) [inline docs/api.md]: This is cumbersome if I want to keep my existing event-based code working with minimal changes. Do I have to replace `page.setRequestInterception` with `page.route('*', () =>{})` ? Maybe we can support `page.route()` call for that (also imperfect?

**Review by yury-s — APPROVED** [review]
I like the `page.route()` API much better but if we need to keep the old event-based handling too we should think more about how they would work together.

File: docs/api.md (hunk 1 above)
Comment (pavelfeldman) [reply docs/api.md]: There is no real migration yet, we are at 0.10, but in terms of Puppeteer parity, you delete the `page.setRequeatInterception` call and replace `page.on` with `page.route` instead. Empty `page.route()` does not make sense - it stalls the world.

I don't expect `request.continue` to be called from event ever...

**Comment by lostintangent** [comment]
I love this change! ❤️ Would `interceptRequest` be another alternative name worth considering? I'm just wondering if `route` conveys the intent of the code as clearly as it could. Maybe I've just done too much Express/etc. development, but it reads like you're implementing a web server, as opposed to intercepting specific network requests.

**Review by aslushnikov — COMMENTED** [review]
`page.route` is great!

**Comment by pavelfeldman** [comment]
> Would `interceptRequest()` be another alternative

@lostintangent: In some sense I wanted to give express users a hint on what that is, in the end of the day, just like the server, you serve that request there. `interceptRequest` would work, but sounds a bit scary to me.

**Pull request merged**